src/mutation_data.cpp

```diff
@@ -641,7 +641,21 @@ void dream::load( const JsonObject &jsobj )
 
 bool trait_display_sort( const trait_id &a, const trait_id &b ) noexcept
 {
-    return a->name() < b->name();
+    if( a->get_display_color() > b->get_display_color() ) {
+        return true;
+    }
+    if( a->get_display_color() < b->get_display_color() ) {
+        return false;
+    }
+
+    if( a->name() < b->name() ) {
+        return true;
+    }
+    if( a->name() > b->name() ) {
+        return false;
+    }
```

Comment:
FWIW I'd write this as
```
return std::make_pair( a->get_display_color(), a->get_name() ) <
       std::make_pair( b->get_display_color(), b->get_name() );
```

Reply:
I totally would do it this way if both sort criteria were <.  get_display_color() is > while the name is <. This forces the display colors to be yellow/green/red, then sorted alphabetically (A-Z). Doing it by make_pair comparison would result in Red/Green/Yellow sorted A-Z, or Yellow/Green/Red sorted Z-A. Definitely would have been an elegant solution though

Reply:
Tried it since I liked how it's written. Here's the result after flipping the comparison operand to make the colors match up

![image](https://user-images.githubusercontent.com/63547126/80130315-5f490200-8566-11ea-96dd-384d967bf8fd.png)

Reply:
Oh well, that's a shame.  I think you could negate each call to `get_display_color()`; since they're implicitly convertible to an `int` that should work.  That might be more confusing to read, though.